### PR TITLE
Added support for secret description

### PR DESCRIPTION
**Pull request by martinbrose**
Addresses:

- #1556

### CODE REVIEW

**Review by Wauplin — CHANGES_REQUESTED** [review]
Thanks for taking care of this @martinbrose ! I've added a comment about empty strings vs None values. Otherwise PR looks good to me. Thanks for taking care of documentation + tests!

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -4975,7 +4975,9 @@ def hide_discussion_comment(
         return deserialize_event(resp.json()["updatedComment"])  # type: ignore
 
     @validate_hf_hub_args
-    def add_space_secret(self, repo_id: str, key: str, value: str, *, token: Optional[str] = None) -> None:
+    def add_space_secret(
+        self, repo_id: str, key: str, value: str, description: Optional[str] = str(), *, token: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (martinbrose) [reply src/huggingface_hub/hf_api.py]: Thanks, makes sense. Changed.

**Review by Wauplin — APPROVED** [review]
Looks good to me! Let's wait to see if the CI gets green, and it'll be good to merge :)

**Comment by Wauplin** [comment]
Some errors in the CI but not related to your work @martinbrose. Merging this one right now. Great job! :tada:

**Pull request merged**